src/executor/CsvExporter.py

```diff
@@ -4,21 +4,70 @@
 from ..utils import get_project_root
 import os
 import csv
+import re
+import numpy as np
 from pathlib import Path
+import ffmpeg
 
-STORAGE_DIR = Path(os.path.join(get_project_root(), "data_files"))
-STORAGE_DIR.mkdir(parents=True, exist_ok=True)
+STORAGE_DIR_POSITIVES = Path(os.path.join(get_project_root(), "data_files"))
+STORAGE_DIR_POSITIVES.mkdir(parents=True, exist_ok=True)
+STORAGE_DIR_NEGATIVES = Path(os.path.join(get_project_root(), "data_files"))
+STORAGE_DIR_NEGATIVES.mkdir(parents=True, exist_ok=True)
 
 class CsvExporter(iExecutor):
+    def __init__(self, *parents, target_fps=20, clip_length='5s', length_threshold='3s'):
+        super().__init__(*parents)
+        self.target_fps = target_fps
+        self.clip_len_s = CsvExporter.parse_time(clip_length)
+        self.len_thresh_s = CsvExporter.parse_time(length_threshold)
+
+    def parse_time(time):
+        pattern = r'(?:(\d+)h)?(?:(\d+)m)?(?:(\d+)s)fba3def14910ccdce015f36a03524dd679ba58?'
+        result = re.match(patern, time)
+        if not result:
+            raise ValueError(f'Invalid time: {time}. Expected digit followed by [smh]')
+        hours, minutes, seconds = result.groups()
+        time_s = hours or 0
+        time_s *= 60
+        time_s += minutes or 0
+        time_s *= 60
+        time_s += seconds
+        if time_s <= 0:
+            raise ValueError(f'Invalid time: {time}. Expected a non-zero positive value')
+        return time_s
+
     def run(self, item: VideoItem):
         print("Start exporting file")
         metadata = iExecutor.get_metadata(item)
         bbs = metadata.bb_fields
         if bbs is None or len(bbs) == 0:
             raise StopSignal("No bounding box fields for this item")
 
+        # First, we trim the last seconds before the first collision
+        collision_frame = np.min(item['accident_locations'])
```

Comment:
This will fail if there are no accidents right? Guess this is why you were asking about the filter condition. Is this something we want to handle?